geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/scala/org/locationtech/geomesa/fs/storage/common/StorageMetadata.scala

```diff
@@ -342,14 +343,16 @@ object StorageMetadata extends MethodProfiling with LazyLogging {
     * @param root root path
     * @param config config
     */
-  private def writeStorageConfig(fc: FileContext, root: Path, config: StorageConfig): Unit = {
+  private def writeStorageConfig(fc: FileContext, root: Path, config: StorageConfig, charset: Option[String] = Option("UTF-8")): Unit = {
     val file = new Path(root, MetadataPath)
     val data = profile("Serialized storage configuration") {
       ConfigWriter[StorageConfig].to(config).render(options)
     }
     profile("Persisted storage configuration") {
       WithClose(fc.create(file, java.util.EnumSet.of(CreateFlag.CREATE), CreateOpts.createParent)) { out =>
-        out.writeBytes(data)
+        charset.map(c => {
+          out.write(data.getBytes(c))
+        }).getOrElse(out.writeBytes(data))
```

Comment:
it would be better to use a `match` here, since the return value from `.map` is not used:
```
charset match {
  case None => out.writeBytes(data)
  case Some(c) => out.writeBytes(data.getBytes(c))
}
```
I noticed that there is also a `writeUTF` method, though - I wonder if there is some difference in just writing the raw bytes, as the `writeUTF` method also writes the byte length